bucket-cache.go

```diff
@@ -165,7 +165,13 @@ func (c Client) getBucketLocationRequest(bucketName string) (*http.Request, erro
 
 	// Set sha256 sum for signature calculation only with signature version '4'.
 	if c.signature.isV4() {
-		req.Header.Set("X-Amz-Content-Sha256", hex.EncodeToString(sum256([]byte{})))
+		var contentSha256 string
+		if c.secure {
+			contentSha256 = "UNSIGNED-PAYLOAD"
```

Comment:
This is part of the s3 protocol and it doesn't seem that it will be changed anytime soon :)